apps/server/src/modules/system/service/system.service.ts

```diff
@@ -15,13 +16,17 @@ export class SystemService {
 		} else {
 			systemEntities = await this.systemRepo.findByFilter(type, onlyOauth);
 		}
-		const systemDtos: SystemDto[] = SystemMapper.mapFromEntitiesToDtos(systemEntities);
-		return systemDtos;
+		const keycloakConfig = (await this.systemRepo.findByFilter(SysType.KEYCLOAK, true))[0];
+		systemEntities.forEach((systemEntity) => {
+			if (keycloakConfig && systemEntity.type === SysType.OIDC && !systemEntity.oauthConfig) {
```

Comment:
So if `keycloak` is falsy you can skip the entire `forEach`, right?

Reply:
Yes, if keycloakConfig is falsy, no keycloakConfig.oauthConfig can be assigned to the systemEntities. So the forEach loop can be skipped. But in practice there should always exists a keycloak.oauthConfig, so I removed the condition from the if query.